Testscripts/Linux/validate-da.sh

```diff
@@ -180,53 +195,65 @@ function enable_disable_da(){
 
     # Wait for 30s for DA to start running and check for service.log
     sleep 30
-    verify_expected_file "/var/opt/microsoft/dependency-agent/log/service.log"
+    verify_expected_file "$da_log_dir/service.log"
 
     # Check for service.log.1 file and confirm service.log.2 doesn't exist
-    service_log_time=$(date -r /var/opt/microsoft/dependency-agent/log/service.log +%s)
-    time_limit=$(($service_log_time + 60))
-    if [ "$DISTRO" == "debian"*] || ["$DISTRO" == "ubuntu"* ]; then
-        time_limit=$(($time_limit + 60))
-    fi
+    service_log_time=$(date -r $da_log_dir/service.log +%s)
+    time_limit=$(($service_log_time + $enable_timeout))
 
-    while [ ! -f "/var/opt/microsoft/dependency-agent/log/service.log.1" ]
+    while [ ! -f "$da_log_dir/service.log.1" ]
     do
         current_time=$(date +%s)
         if [ $current_time -gt $time_limit]; then
-            SetTestStateFailed
-            return
+            fail_test "Exceeded time limit to check service.log.1 file"
         fi
         sleep 5
     done
 
-    verify_expected_file "/var/opt/microsoft/dependency-agent/log/service.log.1"
-    [ -f "/var/opt/microsoft/dependency-agent/log/service.log.2" ] && fail_test "/var/opt/microsoft/dependency-agent/log/service.log.2 exist."
-    [ ! -c "/dev/msda" ] && fail_test "/dev/msda does not exist."
-
+    verify_expected_file "$da_log_dir/service.log.1"
+    
+    if [ -f "$da_log_dir/service.log.2" ]; then
+        fail_test "$da_log_dir/service.log.2 exist."
+    fi
+    
+    if [ ! -c "/dev/msda" ]; then 
+        fail_test "/dev/msda does not exist."
+    fi
     # Check for "driver setup status=0" and "starting the dependency agent" in service.log.1 and service.log respectively
-    ! grep -iq "driver setup status=0" /var/opt/microsoft/dependency-agent/log/service.log.1 && fail_test "driver setup status=0 not found"
-    ! grep -iq "starting the dependency agent" /var/opt/microsoft/dependency-agent/log/service.log && fail_test starting the dependency agent not found
+    if ! grep -iq "driver setup status=0" $da_log_dir/service.log.1; then
```

Comment:
Request for opinions:

This assertion may fail because either a) the "driver setup status" line is missing completely, or b) the line exists, but the setup status was non-zero.

Question: Does it make sense to differentiate between the two failure modes? Extract the line of text without regard for the setup value, fail if not found. If it was found, then assert that the setup value was zero and include the actual value if the LogErr message if the assertion fails. The actual setup status provides diagnostic data to assist in troubleshooting the failure.

Reply:
I changed the logic to consider non-zero driver setup status as well